buildmultiplescalaversions.sh

```diff
@@ -1,10 +1,20 @@
 #! /bin/bash
+BASEDIR=$(dirname $(readlink -f "$0"))
+
+function whatchanged() {
+    cd $BASEDIR
+    for i in $(git status -s --porcelain -- $(find ./ -mindepth 2 -name pom.xml)|awk '{print $2}'); do
+	echo $(dirname $i)
+	cd $BASEDIR
+    done
+}
+
 set -eu
 ./change-scala-versions.sh 2.11 # should be idempotent, this is the default
 mvn "$@"
 ./change-spark-versions.sh 2
-mvn -Dspark.major.version=2 "$@"
+mvn -Dspark.major.version=2 -Dmaven.clean.skip=true -pl $(whatchanged| tr '\n' ',') -amd "$@"
```

Comment:
Don't need/want ```-Dspark.major.version=2``` here - it doesn't matter with the change-spark-versions.sh earlier, and it wouldn't be flattened anyway (i.e., value would still read from pom, instead of being substituted/hard-coded in).